mkt/search/filters.py

```diff
@@ -102,7 +105,9 @@ def filter_queryset(self, request, queryset, view):
 
         return queryset.query(
             'function_score',
-            query=query.Bool(should=should),
+            query=query.Bool(
+                should=[query.Bool(should=should, minimum_should_match=2),
```

Comment:
I agree. This has too much impact on our search queries. It's way too late but using something related to `function_score` could be an interesting solution here, I agree.